pkg/operator/ceph/config/conditions.go

```diff
@@ -44,46 +45,48 @@ func ConditionExport(context *clusterd.Context, namespace, name string, conditio
 
 // setCondition updates the conditions of the cluster custom resource
 func setCondition(context *clusterd.Context, namespace, name string, newCondition cephv1.Condition) {
-	cluster, err := context.RookClientset.CephV1().CephClusters(namespace).Get(name, metav1.GetOptions{})
-	if err != nil {
-		logger.Errorf("failed to get cluster %v", err)
-	}
-	if conditions == nil {
-		conditions = &cluster.Status.Conditions
-		if cluster.Status.Conditions != nil {
-			conditionMapping(*conditions)
+	if name != "" {
+		cluster, err := context.RookClientset.CephV1().CephClusters(namespace).Get(name, metav1.GetOptions{})
+		if err != nil {
+			logger.Errorf("failed to get cluster %v", err)
```

Comment:
here also you need to return instead of crashing, then you don't need to check for name != ""